teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,465 @@
+process][] which itself seems to be very similar to the [Python PEP
+process][]
+
+[Rust RFC process]: https://github.com/rust-lang/rfcs
+[Kubernetes KEP process]: https://github.com/kubernetes/enhancements/tree/master/keps
+
+
+## Examples
+
+Let's give some example of workflow to give reader a better
+understanding on how and when TEP should be created and how they are
+managed across time.
+
+Those are examples, and do not necessarily reflect what happened, or
+will happens on the particular subject they are about. They are here
+to give more context and ideas on different situation that could rise
+while following the TEP process.
+
+### Share Task and Pipeline as OCI artifact
+
+For more context, this is linked to the following:
+
+- [Feature: Versioning on Tasks/Pipelines](https://github.com/tektoncd/pipeline/issues/1839)
+- [Oci tool: makes use of oci-artifacts to store and retrieve Tekton resources](https://github.com/tektoncd/experimental/pull/461)
+
+1. An initial design doc is crafted (let's imaging it is [Tekton OCI
+Image
+Catalog](https://docs.google.com/document/d/1zUVrIbGZh2R9dawKQ9Hm1Cx3GevKIfOcRO3fFLdmBDc/edit#heading=h.tp9mko2koenr)).
+   An experimental project has already been created and a
+   proof-of-concept demoed during a working group. The next step is to
+   create a TEP (and continue work on the proof-of-concept if need be).
+2. From this design docs, a TEP is created with the content of the
+   design document.
+3. It is approved with a `proposed` state, which means :
+   - We acknowledge this is important for the project, and needs to be
+     worked on
+   - It needs some work and discussion based on the initial proposal
+4. The TEP is being disscussed during Working Group(s) — it can be the
+   main one, a specific one, the API Working Group.
+
+   During those discussion it is clear that some work needs to be
+   done:
+   - Define a Spec for the OCI image (layers, metadata, configuration)
+     The experimental project can be used to demo and validate that spec.
+   - Once the spec is agreed one
+   - Have a new TEP to add support for referencing Task and Pipeline
+     through alternative means than in clusters (OCI image is one,
+     using Git or an HTTP url are others)
+
+   The next action are :
+   - Create a new TEP on support for referencing Task and Pipeline.
+     As before, the TEP can be first discussed during Working group
+     and refined in Google Docs before being proposed as a TEP.
+   - Update the current TEP to define the spec (same thing as above
+     applies). A name is choosed for those : Tekton Bundles.
+   - Create a new TEP on implementing Tekton Bundles in tektoncd
+     projects (`pipeline` and `cli`)
+5. The current TEP, defining the spec, is *approved* and marked as
+   `implemented`. In this case `implemented` means it is available in
+   the documentation in `tektoncd` (most likely on the
+   `tektoncd/pipeline` repository)
```

Comment:
im a bit confused by this - we've still only got TEPs at this point right, no code?

Reply:
> im a bit confused by this - we've still only got TEPs at this point right, no code?

So you have code in `tektoncd/experimental` at this point, you can have code in a fork, or even in a `WIP` PR even ; you just don't have code **merged** into the *non-experimental* repositories (`pipeline`, `cli`, …). But this is about the `spec` TEP, which.. doesn't really need code to be considered as implemented, the "output" of defining a spec is the spec itself (which you could consider as code)